internal/framework/provider/config.go

```diff
@@ -0,0 +1,103 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/mongodb-forks/digest"
+	"github.com/mongodb/terraform-provider-mongodbatlas/version"
+	"github.com/spf13/cast"
+	matlasClient "go.mongodb.org/atlas/mongodbatlas"
+	realmAuth "go.mongodb.org/realm/auth"
+
+	"go.mongodb.org/realm/realm"
+)
+
+type Config struct {
+	PublicKey    string
+	PrivateKey   string
+	BaseURL      string
+	RealmBaseURL string
+	AssumeRole   *AssumeRole
+}
+
+type AssumeRole struct {
+	RoleARN           string
+	Duration          time.Duration
+	ExternalID        string
+	Policy            string
+	PolicyARNs        []string
+	SessionName       string
+	SourceIdentity    string
+	Tags              map[string]string
+	TransitiveTagKeys []string
+}
+
+// MongoDBClient client
+type MongoDBClient struct {
+	Atlas  *matlasClient.Client
+	Config *Config
+}
+
+var ua = "terraform-provider-mongodbatlas/" + version.ProviderVersion
+
+// NewClient func...
+func (c *Config) NewClient(ctx context.Context) (*MongoDBClient, error) {
```

Comment:
Since I had to check the definition of `c` to understand that `c` was `config` in the method 😅 
```suggestion
func (config *Config) NewClient(ctx context.Context) (*MongoDBClient, error) {
```